src/server_manager/install_scripts/install_server.sh

```diff
@@ -406,5 +433,54 @@ ${FIREWALL_STATUS}
 END_OF_SERVER_OUTPUT
 } # end of install_shadowbox
 
-# Wrapped in a function for some protection against half-downloads.
-install_shadowbox
+function read_param() {
+  if (( "$#" < 2 )); then
+    log_error "Missing $1 value"
+    display_usage
+    exit 1
+  fi
+  echo $2
+}
+
+function parse_flags() {
+  local PARAMS=""
+  while [[ "$#" > 0 && ! "$1" == "--" ]]; do
+    case "$1" in
+      --hostname)
+        FLAGS_HOSTNAME=$(read_param "$@")
+        shift
+        ;;
+      --port-for-api)
+        FLAGS_PORT_FOR_API=$(read_param "$@")
+        shift
+        ;;
+      --port-for-keys)
+        FLAGS_PORT_FOR_KEYS=$(read_param "$@")
+        shift
+        ;;
+      -*|--*=) # unsupported flags
+        log_error "Unsupported flag $1"
+        display_usage
+        exit 1
+        ;;
+      *) # preserve positional arguments
+        PARAMS="$PARAMS '$1'"
+        ;;
+    esac
+    shift
+  done
+  if (( FLAGS_PORT_FOR_API == FLAGS_PORT_FOR_KEYS )); then
```

Comment:
Should we check that these are in the valid range (0-65535)?

Reply:
Done